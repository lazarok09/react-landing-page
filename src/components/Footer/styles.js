import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';
import { Container as SectionContainer } from '../SectionContainer/styles';
export const Container = styled.footer`
  ${({ theme }) => css`
    text-align: center;
    font-size: ${theme.font.sizes.small};
    border-top: 0.1rem solid ${theme.colors.mediumGray};
    a {
      color: inherit;
      text-decoration: none;
    }
    & ${TextComponent} {
      font-size: ${theme.font.sizes.small};
      padding-top: 0;
      padding-bottom: 0;
    }
    & ${SectionContainer} {
      padding-top: 0;
      padding-bottom: 0;
    }
  `}
`;
