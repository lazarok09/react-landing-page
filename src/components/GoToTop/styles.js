import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    display: flex;
    justify-content: center;
    align-items: center;
    z-index: 6;
    bottom: 2rem;
    right: 2rem;
    width: 4rem;
    height: 4rem;
    opacity: 0.8;
    color: ${theme.colors.white};
    background: ${theme.colors.primaryColor};
  `}
  &:hover {
    opacity: 1;
  }
`;
