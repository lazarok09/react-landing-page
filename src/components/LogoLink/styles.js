import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    color: inherit;
    display: flex;
    align-items: center;
    text-decoration: none;

    > img {
      max-height: 2.5rem;
    }
  `}
`;
