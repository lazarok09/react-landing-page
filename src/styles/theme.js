export const theme = {
  colors: {
    whiteBg: 'white',
    darkBg: 'black',
  },
};
export default theme;
