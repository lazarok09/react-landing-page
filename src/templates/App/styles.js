import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    background: ${theme.colors.darkBg};
    color: ${theme.colors.colorRed};
  `}
`;
