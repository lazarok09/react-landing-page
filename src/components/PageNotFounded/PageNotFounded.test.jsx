import { PageNotFounded } from './index';
import { screen } from '@testing-library/react';
import { renderTheme } from './../../styles/render-theme';
describe('<PageNotFounded />', () => {
  it('should render PageNotFounded correctly', () => {
    const { container } = renderTheme(<PageNotFounded />);
    const heading = screen.getByRole('heading', { name: /error 404/i });
    const link = screen.getByRole('link', { name: /clique para voltar/i });
    expect(heading).toBeInTheDocument();
    expect(link).toHaveAttribute('href', '/');
    expect(container).toMatchSnapshot();
  });
});
