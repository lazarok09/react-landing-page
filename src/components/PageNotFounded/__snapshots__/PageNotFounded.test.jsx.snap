// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageNotFounded /> should render PageNotFounded correctly 1`] = `
.c3 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c0 {
  background: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
  width: 100%;
}

.c5 {
  font-size: 2.4rem;
}

.c2 {
  text-align: center;
  max-width: 58rem;
  margin: 0 auto;
}

.c4 {
  margin: 6.4rem 0;
}

@media (max-width:768px) {
  .c3 {
    font-size: 4.0rem;
  }
}

<div>
  <div
    class="c0"
    id="id-8440-952173863656"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3"
        >
          Error 404
        </h2>
        <div
          class="c4"
        >
          <div
            class="c5"
          >
            <p>
              A página que você busca não foi encontrada, 
              <a
                href="/"
              >
                clique para voltar
              </a>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
